File: larix/src/main/kotlin/com/haxul/larix/model/Block.kt
package com.haxul.larix.model

import com.haxul.larix.common.GenesisData
import java.time.LocalDateTime

class Block(
    val timestamp: LocalDateTime,
    val lastHash: String,
    val hash: String,
    val data: Any,
    val nonce: Int,
    val difficulty:Int
) {
    companion object {
        val GENESIS_BLOCK: Block = Block(
            GenesisData.TIMESTAMP,
            GenesisData.LAST_HASH,
            GenesisData.HASH,
            GenesisData.DATA,
            GenesisData.NONCE,
            GenesisData.DIFFICULTY
        )
    }

    override fun toString(): String = "Block(timestamp:$timestamp,lastHash:$lastHash,hash:$hash,data:$data)"

}

File: fir/block.ts
class Block {

    timestamp: number
    lastHash: string
    hash: string
    data: any

    constructor(timestamp: number, lastHash: string, hash: string, data: any) {
        this.timestamp = timestamp
        this.lastHash = lastHash
        this.hash = hash
        this.data = data
    }

    public toString = (): string => `
    Block:
       timestamp: ${this.timestamp},
       lastHash: ${this.lastHash},
       hash: ${this.hash},
       data: ${this.data}
    `
}

export default Block